irc/server.go

```diff
@@ -108,7 +119,26 @@ func (s *Server) waitForClientCleanup(cc *clientConnection) {
 	s.Unlock()
 }
 
-// Handle fanning out IRC messages generated from Slack events
+func (s *Server) waitForkillListener(l *net.TCPListener) {
+	<-s.stopChan
+	l.Close()
+
+	// First grab the lock and grab the active connections
+	conns := make([]*clientConnection, 0)
+	s.RLock()
+	for _, conn := range s.clientConnections {
+		conns = append(conns, conn)
+	}
+	s.RUnlock()
+
+	// Now try to close them. This list could be stale, but it won't
+	// cause any deadlocks
+	for _, conn := range conns {
+		conn.shutdown <- true
```

Comment:
`close(conn.shutdown)` my dude

Reply:
incidentally this is why I prefer to make my shutdown channels `chan interface{}` because it reminds me that the value is actually not important at all